run_docker.sh

```diff
@@ -13,108 +12,143 @@
-    read -p "Enter absolute path to Data folder [default: $default_data_dir]: " data_dir
-    read -p "Enter absolute path to store results [default: $default_results_dir]: " results_dir
-    read -p "Would you like to debug C#? (Requires mono debugger attachment) [default: N]: " debugging
+  read -p "Docker image [default: $DEFAULT_IMAGE]: " IMAGE
+  read -p "Path to Lean config.json [default: $DEFAULT_CONFIG]: " CONFIG_PATH
+  read -p "Path to Data directory [default: $DEFAULT_DATA_DIR]: " DATA_DIR
+  read -p "Path to Results directory [default: $DEFAULT_RESULTS_DIR]: " RESULTS_DIR
+  read -p "Path to Python directory [default: $DEFAULT_PYTHON_DIR]: " PYTHON_DIR
+  read -p "Would you like to debug C#? (Requires mono debugger attachment) [default: N]: " DEBUGGING
 fi
 
 #Have to reset IFS for cfg files to work properly
 IFS=" "
 
-if [ -z "$image" ]; then
-    image=$default_image
-fi
-
-if [ -z "$config_file" ]; then
-    config_file=$default_config_file
-fi
-
-if [ -z "$python_dir" ]; then
-    python_dir=$default_python_dir
-fi
-
-if [ ! -f "$config_file" ]; then
-    echo "Lean config file $config_file does not exist"
-    exit 1
-fi
-
-if [ -z "$data_dir" ]; then
-    data_dir=$default_data_dir
+# fall back to defaults on empty input without
+CONFIG_PATH=${CONFIG_PATH:-$DEFAULT_CONFIG}
+DATA_DIR=${DATA_DIR:-$DEFAULT_DATA_DIR}
+RESULTS_DIR=${RESULTS_DIR:-$DEFAULT_RESULTS_DIR}
+IMAGE=${IMAGE:-$DEFAULT_IMAGE}
+PYTHON_DIR=${PYTHON_DIR:-$DEFAULT_PYTHON_DIR}
+
+# convert to absolute paths
+CONFIG_PATH=$(absolute_path "${CONFIG_PATH}")
+PYTHON_DIR=$(absolute_path "${PYTHON_DIR}")
+DATA_DIR=$(absolute_path "${DATA_DIR}")
+RESULTS_DIR=$(absolute_path "${RESULTS_DIR}")
+CSHARP_DLL=$(absolute_path "${CSHARP_DLL}")
+CSHARP_PDB=$(absolute_path "${CSHARP_PDB}")
+
+if [ ! -f "$CONFIG_PATH" ]; then
+  echo "Lean config file $CONFIG_PATH does not exist"
+  exit 1
 fi
 
-if [ ! -d "$data_dir" ]; then
-    echo "Data directory $data_dir does not exist"
-    exit 1
+if [ ! -d "$DATA_DIR" ]; then
+  echo "Data directory $DATA_DIR does not exist"
+  exit 1
 fi
 
-if [ -z "$results_dir" ]; then
-    results_dir=$default_results_dir
+if [ ! -d "$RESULTS_DIR" ]; then
+  echo "Results directory $RESULTS_DIR does not exist; creating it now"
+  mkdir $RESULTS_DIR
 fi
 
-if [ ! -d "$results_dir" ]; then
-    echo "Results directory $results_dir does not exist"
-    exit 1
-fi
-
-#First part of the docker command that is static, then we build the rest
-command="docker run --rm --mount type=bind,source=$config_file,target=/Lean/Launcher/config.json,readonly \
-    --mount type=bind,source=$data_dir,target=/Data,readonly \
-    --mount type=bind,source=$results_dir,target=/Results \
-    --name LeanEngine \
+#First part of the docker COMMAND that is static, then we build the rest
+COMMAND="docker run --rm -v $CONFIG_PATH:/Lean/Launcher/config.json:ro \
+    -v $DATA_DIR:/Data:ro \
+    -v $RESULTS_DIR:/Results \
```

Comment:
I think we should normalize the behavior across all docker scripts, whether we use `-v` or `--mount`